sources/api/netdog/src/net_config/v1.rs

```diff
@@ -0,0 +1,274 @@
+pub(crate) struct NetConfigV1 {
+    // Use an IndexMap to preserve the order of the devices defined in the net.toml.  The TOML
+    // library supports this through a feature making use of IndexMap.  Order is important because
+    // we use the first device in the list as the primary device if the `primary` key isn't set for
+    // any of the devices.
+    //
+    // A custom type is used here that will ensure the validity of the interface name as according
+    // to the criteria in the linux kernel.  See the `interface_name` module for additional details
+    // on the validation performed.
+    #[serde(flatten)]
+    pub(crate) interfaces: IndexMap<InterfaceName, NetInterfaceV1>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct NetInterfaceV1 {
+    // Use this interface as the primary interface for the system
+    pub(crate) primary: Option<bool>,
+    pub(crate) dhcp4: Option<Dhcp4ConfigV1>,
+    pub(crate) dhcp6: Option<Dhcp6ConfigV1>,
+}
+
+impl Interfaces for NetConfigV1 {
+    fn primary_interface(&self) -> Option<String> {
+        self.interfaces
+            .iter()
+            .find(|(_, v)| v.primary == Some(true))
+            .or_else(|| self.interfaces.first())
+            .map(|(n, _)| n.to_string())
+    }
+
+    fn has_interfaces(&self) -> bool {
+        !self.interfaces.is_empty()
+    }
+
+    fn into_wicked_interfaces(&self) -> Vec<WickedInterface> {
+        let mut wicked_interfaces = Vec::with_capacity(self.interfaces.len());
+        for (name, config) in &self.interfaces {
+            let wicked_dhcp4 = config.dhcp4.clone().map(WickedDhcp4::from);
+            let wicked_dhcp6 = config.dhcp6.clone().map(WickedDhcp6::from);
+            let wicked_interface = WickedInterface {
+                name: name.clone(),
+                control: WickedControl::default(),
+                ipv4_dhcp: wicked_dhcp4,
+                ipv6_dhcp: wicked_dhcp6,
+            };
+            wicked_interfaces.push(wicked_interface)
+        }
+
+        wicked_interfaces
+    }
+}
+
+impl Validate for NetConfigV1 {
+    fn validate(&self) -> Result<()> {
+        let dhcp_misconfigured = self
+            .interfaces
+            .values()
+            .any(|cfg| cfg.dhcp4.is_none() && cfg.dhcp6.is_none());
+        ensure!(
+            !dhcp_misconfigured,
+            error::InvalidNetConfigSnafu {
+                reason: "each interface must configure dhcp4 or dhcp6, or both",
+            }
+        );
+
+        let primary_count = self
+            .interfaces
+            .values()
+            .filter(|v| v.primary == Some(true))
+            .count();
+        ensure!(
+            primary_count <= 1,
+            error::InvalidNetConfigSnafu {
+                reason: "multiple primary interfaces defined, expected 1"
+            }
+        );
+
+        Ok(())
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
```

Comment:
Nothing has really changed from here down.   This code was moved here into `v1.rs` since we use the `NetConfigV1` struct for the simple configuration we get from the kernel command line.